CMakeLists.txt

```diff
@@ -22,19 +20,27 @@ endif()
 if(NOT BUILD_COMM_TCP_PLAIN)
     set(BUILD_COMM_TCP_PLAIN FALSE)
 endif()
+if(NOT BUILD_COMM_TCP_TLS)
+    set(BUILD_COMM_TCP_TLS FALSE)
+endif()
```

Comment:
I think we should add a check right after this, to make sure that at most one of these is set, and possibly also to log which communication system was chosen. It's not until the internal build scripts, or even the code, that it's obvious that only one communication system can be chosen at a time, and that it must be chosen at build time, otherwise.

Reply:
ITs logged in the ConnectionFactory, but of course we can print it from here as well. Someday the comm system will come from configuration - and this flag will be used for build purpose only. But for now you are right, its not clear. WIll add it.

Reply:
@upshaw-alex , sorry for being late for your comment. In general, the decision of which comm module will be used in the runtime will come from configuration, some day. The purpose of these flags is just to build the correct module. But for now, we use these flag to decide both which module to build and what will be the actual comm module for the runtime.